akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -268,6 +268,40 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
+  private var buf = immutable.Stack.empty[T]
```

Comment:
`Vector` should be a better match here unless I'm missing something.
You can also give it a size hint (equal to `n`) once the previous "chunk" has been signalled to avoid it having to resize.